### DISCUSSION TITLE
How can i configure web3auth modal to preset user email field?

### DISCUSSION THREAD

**Discussion by devsalman247** [Web3Auth/Web3Auth#1210] [Help]
Hi there! I am working on a project in which I need to develop an authentication process where the user logs in with google OAuth and when the user has to connect his wallet we've to connect him with his wallet using the email-passwordless option but we don't want to enable user manually enter email field so that user log in with the same email that he has used for signing in with google..
but after reading the docs I still can't see any config option available for this purpose?

**Comment by htcarr3** [comment]
I have a similar situation, would love to understand what is possible.

**Comment by MarcosGin** [comment]
Hello! If I understood correctly, you need to obtain the provider's email, correct?

After having made the connect, you have access to this async method to obtain the user's information, which in addition to the email returns other useful fields.

`const user = await web3Auth.getUserInfo();`

docs: https://web3auth.io/docs/sdk/web/core/usage#get-users-information

**Reply by htcarr3** [reply]
I believe OP (and I) are asking how to prefill the email for the user (the email is received from some other means of authentication)

**Reply by devsalman247** [reply]
@htcarr3 Yes, that's what I'm trying to achieve...

**Comment by arch1995** [comment]
hi @devsalman247 @htcarr3 

Do you need to pre-fill the user email in the web3auth plug-and-play modal?

**Reply by devsalman247** [reply]
Yes @arch1995

**Reply by htcarr3** [reply]
I think the solution is to use the core sdk connectTo() instead of the modal
`      const provider = await web3Auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
        loginProvider: 'email_passwordless',
        mfaLevel: 'none',
        extraLoginOptions: {
          login_hint: 'foo@bar.com'
        }
      })
`
This works for me, but I have issues reconnecting on page reload

**Reply by devsalman247** [reply]
Yeah that should done the work. I'll definitely try it..Thanks!
i've issues too on page reload probably slow internet causing modal init not to initialize properly....

**Reply by arch1995** [reply]
yes, that's correct @htcarr3
